### PR TITLE
Split diplomacy into the standalone repository

### PR DESCRIPTION
**Pull request by sequencer**
submodule currently locate in https://github.com/sequencer/diplomacy/tree/compile_rc

Since diplomacy can be a great framework for parameterized circuit construction, Chisel user depending on rocket-chip will be a burden, if them only wanna use a small framework, rather than entire rocket core.

This PR splits out `LazyModule` `Nodes` `CDE` from diplomacy into a new standalone package, add compatibility layer to rocket-chip, make it be able to use this package as its dependency. 

# TODO

- [x] compile
- [x] fix for sbt compiling.
- [x] pass CI.
- [ ] move diplomacy package to chipsalliance.
- [ ] pass CI.

## In next PRs
- [ ] Remove deprecated warning introduced by this PR

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**
Use standalone diplomacy package.

### CODE REVIEW

**Comment by sequencer** [comment]
only failed in wake build.
I think it can be reviewed now.

File: build.wake [hunk 1]
```diff
@@ -11,17 +11,21 @@ global def hardfloatScalaModule =
...
-  | addMacrosParadiseCompilerPlugin
+def diplomacyRoot = findGitRepositoryWithFallback "diplomacy"
+
+global def diplomacyMacrosModule =
+  makeScalaModuleFromJSON here "diplomacyMacros"
+  | setScalaModuleRootDir "{diplomacyRoot}/diplomacy/macros"
+
+global def diplomacyModule =
+  def deps = diplomacyMacrosModule, Nil
+  makeScalaModuleFromJSON here "diplomacy"
```
Comment (jackkoenig) [inline build.wake:22]: This is saying there's a key "diplomacy" in `ivydependencies.json`. You need to add one to get the tests using Wake to pass.

Comment (sequencer) [reply build.wake:22]: Got it! Thank you, I think I used to add it, but seems to be dropped in a rebasing.

**Comment by sequencer** [comment]
API change see:
add
```
getInfo getParent getDiParams getUiParams getUoParams getDoParams
```
to access correspond `val`s 
change 
```
protected[diplomacy] val dangles
```
to 
```
val dangles
```

**Comment by sequencer** [comment]
Request review from @jackkoenig @hcook @terpstra. Thank you!

**Comment by jerryz123** [comment]
This is done now

**Pull request closed without merging**